File: temp_converter.py
def celsius_to_fahrenheit(c):
    return c * 9 / 5 + 32


def celsius_to_kelvin(c):
    return c + 273.15


def main():
    print("Конвертер температуры (C, F, K)")
    print("Доступные шкалы: C, F, K")

    value = float(input("Введите значение: "))
    unit = input("Введите исходную шкалу (C/F/K): ").strip().upper()

    if unit == "C":
        f = celsius_to_fahrenheit(value)
        k = celsius_to_kelvin(value)
        print(f"{value}°C = {f:.2f}°F = {k:.2f}K")
    elif unit == "F":
        c = (value - 32) * 5 / 9
        k = celsius_to_kelvin(c)
        print(f"{value}°F = {c:.2f}°C = {k:.2f}K")
    elif unit == "K":
        c = value - 273.15
        f = celsius_to_fahrenheit(c)
        print(f"{value}K = {c:.2f}°C = {f:.2f}°F")
    else:
        print("Ошибка: неизвестная шкала")
        return

    # Таблица от -50 до +50 по Цельсию
    print("\nТаблица перевода (Цельсий -> Фаренгейт, Кельвин):")
    print("  C     F     K")
    for c in range(-50, 51, 10):
        f = celsius_to_fahrenheit(c)
        k = celsius_to_kelvin(c)
        print(f"{c:4} {f:5.1f} {k:5.1f}")


if __name__ == "__main__":
    main()

File: temp_converter.cpp
#include <iostream>
#include <iomanip>
#include <string>

double celsius_to_fahrenheit(double c) {
    return c * 9.0 / 5.0 + 32;
}

double celsius_to_kelvin(double c) {
    return c + 273.15;
}

int main() {
    setlocale(LC_ALL, "ru");
    std::cout << "Конвертер температуры (C, F, K)" << std::endl;
    std::cout << "Доступные шкалы: C, F, K" << std::endl;

    double value;
    char unit;

    std::cout << "Введите значение: ";
    std::cin >> value;
    std::cout << "Введите исходную шкалу (C/F/K): ";
    std::cin >> unit;

    if (unit == 'C' || unit == 'c') {
        double f = celsius_to_fahrenheit(value);
        double k = celsius_to_kelvin(value);
        std::cout << value << "°C = " << f << "°F = " << k << "K" << std::endl;
    }
    else if (unit == 'F' || unit == 'f') {
        double c = (value - 32) * 5.0 / 9.0;
        double k = celsius_to_kelvin(c);
        std::cout << value << "°F = " << c << "°C = " << k << "K" << std::endl;
    }
    else if (unit == 'K' || unit == 'k') {
        double c = value - 273.15;
        double f = celsius_to_fahrenheit(c);
        std::cout << value << "K = " << c << "°C = " << f << "°F" << std::endl;
    }
    else {
        std::cout << "Ошибка: неизвестная шкала" << std::endl;
        return 1;
    }

    std::cout << "\nТаблица перевода (Цельсий -> Фаренгейт, Кельвин):" << std::endl;
    std::cout << "  C     F     K" << std::endl;
    for (int c = -50; c <= 50; c += 10) {
        double f = celsius_to_fahrenheit(c);
        double k = celsius_to_kelvin(c);
        std::cout << std::setw(4) << c << std::setw(6) << std::fixed << std::setprecision(1) << f << std::setw(6) << k << std::endl;
    }

    return 0;
}